clang/lib/AST/Interp/ByteCodeExprGen.cpp

```diff
@@ -884,8 +884,36 @@ bool ByteCodeExprGen<Emitter>::visitInitList(ArrayRef<const Expr *> Inits,
     if (!this->emitDupPtr(E))
       return false;
 
+    // guard relatively expensive base check behind an almost-always-false
```

Comment:
The comment sounds like this is just a performance improvement. Is that so?

Reply:
Maybe half so? It's an attempt to avoid a performance regression that I think I would've otherwise introduced by making this change.